src/promptflow/tests/sdk_cli_azure_test/recording_utilities/bases.py

```diff
@@ -131,7 +142,22 @@ def _process_response_recording(self, response: Dict) -> Dict:
             for processor in self.recording_processors:
                 response = processor.process_response(response)
 
-        response["body"]["string"] = response["body"]["string"].encode("utf-8")
+        if is_async_response:
+            response["content"] = response["body"]["string"]
+            if not is_replay():
```

Comment:
Why do we need to do this?

Reply:
in short, httpx response has different contract with others network pkgs, so we need some different logic to handle.

Reply:
in this place, non-replay mode means record mode (as live will directly return at the very beginning)
then in record mode, response need to be returned as what it looks like as httpx response, which means, string type content, and no body.string; otherwise, it will be recorded in recording file as duplication; so in this case, we need to pop body part, and promise the string type.